R/Rules-class.R

```diff
@@ -448,6 +448,77 @@ NextBestInfTheory <- function(target, asymmetry) {
   .NextBestInfTheory(target = target, asymmetry = asymmetry)
 }
 
+# NextBestTDsamples ----
+
+## class ----
+
+#' `NextBestTDsamples`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`NextBestTDsamples`] is the class to find a next best dose based on Pseudo
+#' DLT model with samples. Namely, it is to find two next best doses, one
+#' for allocation during the trial and the second for final recommendation at
+#' the end of a trial. Hence, there are two target probabilities of the
+#' occurrence of a DLT that must be specified: target probability to be used
+#' during the trial and target probability to be used at the end of the trial.
+#'
+#' @slot prob_target_drt (`proportion`)\cr the target probability of the
+#'   occurrence of a DLT to be used during the trial.
+#' @slot prob_target_eot (`proportion`)\cr the target probability of the
+#'   occurrence of a DLT to be used at the end of the trial.
+#' @slot derive (`function`)\cr the function which derives, based on a vector of
+#'   the posterior dose samples, the target dose, which has the probability of
+#'   the occurrence of DLT equals to either the `prob_target_drt` or
+#'   `prob_target_eot`. It must therefore accept one and only one argument,
+#'   which is a numeric vector, and return a number.
+#'
+#' @aliases NextBestTDsamples
+#' @export
+#'
+.NextBestTDsamples <- setClass(
+  Class = "NextBestTDsamples",
+  slots = c(
+    prob_target_drt = "numeric",
+    prob_target_eot = "numeric",
+    derive = "function"
+  ),
+  prototype = prototype(
+    prob_target_drt = 0.35,
+    prob_target_eot = 0.3,
+    derive = function(dose_samples) {
+      quantile(dose_samples, prob = 0.3)
+    }
+  ),
+  contains = "NextBest",
+  validity = v_next_best_td_samples
+)
+
+## constructor ----
+
+#' @rdname NextBestTDsamples-class
+#'
+#' @param prob_target_drt (`proportion`)\cr the target probability of the
+#'   occurrence of a DLT to be used during the trial.
+#' @param prob_target_eot (`proportion`)\cr the target probability of the
```

Comment:
do we always repeat all the documentation in the constructor? maybe we could just say generically `see slot definition` ?

Reply:
Yes, we do, unfortunately :( I was also thinking about this today, I agree, it would be better to use `see slot definition`.

Reply:
I will clean this for `Rules`, and for separates files that were already cleaned up, I will create a new issue.